code/datums/craft/gun_parts.dm

```diff
@@ -21,6 +21,40 @@
 	var/barrel = /obj/item/part/gun/barrel
 	var/barrel_attached = FALSE
 
+/obj/item/part/gun/frame/New(loc)
+	..()
+	var/spawn_with_preinstalled_parts = FALSE
+	if(istype(loc, /obj/structure/scrap_spawner))
+		spawn_with_preinstalled_parts = TRUE
+	else if(in_maintenance())
+		var/turf/T = get_turf(src)
+		for(var/atom/A in T.contents)
+			if(istype(A, /obj/spawner))
+				spawn_with_preinstalled_parts = TRUE
+
+	if(spawn_with_preinstalled_parts)
+		var/list/parts_list = list(pick(gripvars), mechanism, barrel)
+
+		parts_list -= pick(parts_list)
+		if(prob(50))
+			parts_list -= pick(parts_list)
+
+		for(var/part in parts_list)
+			if(ispath(part, grip))
+				new part(src)
+				grip_attached = TRUE
+			else if(part in gripvars)
+				var/variantnum = gripvars.Find(part)
+				result = resultvars[variantnum]
+				new part(src)
+				grip_attached = TRUE
```

Comment:
Could you explain why this is needed? 
part is picked from `list(pick(gripvars), mechanism, barrel)` so it's either a grip, mechanism or barrel. Or do you know a case where gripvars do not contain only items of type `/obj/item/part/gun/grip`?

Reply:
It's not needed with the current gun frames, but I wanted to have something in case a variant grip was added when a specific grip was defined. Say, the grip is `/obj/item/part/gun/grip/wood`, but `/obj/item/part/gun/grip/rubber` is a variant. It would fail that first check and not add anything.